medusa/clients/torrent/generic.py

```diff
@@ -44,67 +44,30 @@ def __init__(self, name, host=None, username=None, password=None):
         self.username = app.TORRENT_USERNAME if username is None else username
         self.password = app.TORRENT_PASSWORD if password is None else password
         self.host = app.TORRENT_HOST if host is None else host
+        self.torrent_path = app.TORRENT_PATH if torrent_path is None else torrent_path
         self.rpcurl = app.TORRENT_RPCURL
         self.url = None
         self.response = None
         self.auth = None
-        self.last_time = time.time()
+        self.message = None
         self.session = ClientSession()
         self.session.auth = (self.username, self.password)
-
-    def _request(self, method='get', params=None, data=None, files=None, cookies=None):
-
-        if time.time() > self.last_time + 1800 or not self.auth:
-            self.last_time = time.time()
-            self._get_auth()
-
-        text_params = str(params)
-        text_data = str(data)
-        text_files = str(files)
-        log.debug(
-            '{name}: Requested a {method} connection to {url} with'
-            ' params: {params} Data: {data} Files: {files}', {
-                'name': self.name,
-                'method': method.upper(),
-                'url': self.url,
-                'params': text_params[0:99] + '...' if len(text_params) > 102 else text_params,
-                'data': text_data[0:99] + '...' if len(text_data) > 102 else text_data,
-                'files': text_files[0:99] + '...' if len(text_files) > 102 else text_files,
-            }
-        )
-
-        if not self.auth:
-            log.warning('{name}: Authentication Failed', {'name': self.name})
+        self.verify = certifi.where()
+        
+        if self._get_auth():
             return False
 
-        try:
-            self.response = self.session.request(method, self.url, params=params, data=data, files=files,
-                                                 cookies=cookies, timeout=120, verify=False)
-        except (requests.exceptions.MissingSchema, requests.exceptions.InvalidURL) as error:
```

Comment:
What is your reasoning for removing the exception handling?

Reply:
The function makes use of the ClientSession as defined in medusa.session.core.
The exception handling is done there, so there will never be any exception and this code is useless.
The only error handling to be needed is when you get a normal response but an status code higher or equal 400, but that is not an exception

Reply:
Your absolutely right. Totally forgot that I used the MedusaSafeSession for this.
Thanks for noticing.